index.js

```diff
@@ -108,7 +108,7 @@ class Main {
 
       // Check error here so we can retry if hit rate limit.
       // Reddit returns code 200 for rate limit for some reason.
-      if (r.json.errors.length) {
+      if (r && r.json && r.json.errors && r.json.errors.length) {
```

Comment:
I'm not sure if this would help fix the error you reported. Since it logged `Cannot read property 'errors' of undefined`, `r.json` would be undefined and the following `result.json.data` would fail. This is strange though, perhaps you can try hitting the `/api/submit` endpoint manually using [curl](https://github.com/reddit-archive/reddit/wiki/OAuth2-Quick-Start-Example) and see what reddit gives?